packages/x-data-grid/src/hooks/features/dimensions/useGridDimensions.ts

```diff
@@ -302,7 +302,7 @@ export function useGridDimensions(
   }, [apiRef, savedSize, updateDimensions]);
 
   const root = apiRef.current.rootElementRef.current;
-  const dimensions = apiRef.current.state.dimensions;
+  const dimensions = useGridSelector(apiRef, gridDimensionsSelector);
```

Comment:
This has to be done like this for proper reactivity. Any guesses why it was done in a non-reactive manner previously?

Reply:
No idea. This would probably fit better at the start of the hook with the other state/setter values.